### PR TITLE
WFS optimisations - framework changes to have partition filters in fetcher

### PR DESCRIPTION
**Pull request by sowmya-dixit**

### CODE REVIEW

File: analytics-core/src/main/scala/org/ekstep/analytics/framework/DataFetcher.scala [hunk 1]
```diff
@@ -71,4 +76,21 @@ object DataFetcher {
         null;
     }
 
+    def getFilteredKeys(query: Query, keys: Array[String], partitions: Option[List[Int]]): Array[String] = {
+        if (partitions.nonEmpty) {
+            val dates = CommonUtil.getQueryDates(query)
```
Comment (SanthoshVasabhaktula) [inline analytics-core/src/main/scala/org/ekstep/analytics/framework/DataFetcher.scala:81]: Dates check is redundant. Can we just do regex match for the position 11,12,13 index position?

File: analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala [hunk 2]
```diff
@@ -719,4 +719,16 @@ object CommonUtil {
...
+    val endDate = query.endDate
+
+    val from = if (query.delta.nonEmpty) getStartDate(endDate, query.delta.get) else startDate;
+    if (from.nonEmpty) {
+      getDatesBetween(from.get, endDate)
+    } else {
+      Array(endDate.get)
+    }
+  }
+
```
Comment (SanthoshVasabhaktula) [inline analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala:733]: This method should not be necessary

File: analytics-core/src/main/scala/org/ekstep/analytics/framework/DataFetcher.scala (hunk 1 above)
Comment (sowmya-dixit) [reply analytics-core/src/main/scala/org/ekstep/analytics/framework/DataFetcher.scala]: Done - updated to use regex match for {date}-{partition}

File: analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala (hunk 2 above)
Comment (sowmya-dixit) [reply analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala]: Removed this method

**Pull request merged**